tests/EndToEndTests.hs

```diff
@@ -394,6 +413,25 @@ tests = testSpec "End-to-end tests" $ do
                   -- a typeclass for client-friendly "Show" (separate from
                   -- actual Show which remains extremely useful for debugging)
                   -- and use that when including values in error messages.
+                  [ "message" .= ("Could not coerce Name {unName = \"dogCommand\"} to valid value: ValueScalar' (ConstString (String \"cow\")) not an enum: [Right (Name {unName = \"Sit\"}),Right (Name {unName = \"Down\"}),Right (Name {unName = \"Heel\"})]" :: Text)
+                  ]
+                ]
+              ]
+        toJSON (toValue response) `shouldBe` expected
+      it "Errors when no variables provided" $ do
+        response <- executeQuery  @QueryRoot (rootHandler mortgage) query Nothing mempty
+        let expected =
+              object
+              [ "data" .= object
+                [ "dog" .= object
+                  [ "name" .= ("Mortgage" :: Text)
+                  , "doesKnowCommand" .= Null
+                  ]
+                ]
+              , "errors" .=
+                [
+                  object
+                  -- TODO: cf previous test case
```

Comment:
Not sure I could do this todo. Please either do it or expand on it so that someone who's not you could do it.

Reply:
So this is just a reference to the one above that I did not write. I'm not sure I understand the initial remark which stated:
```
-- TODO: This error message is pretty bad. We should define
-- a typeclass for client-friendly "Show" (separate from
-- actual Show which remains extremely useful for debugging)
-- and use that when including values in error messages.
```

Now, while I understand both the idea and how to do it (I think...), it seems like a lot of work for this PR, since both error messages are pretty close, I thought it would be better to add the TODO reference to the other one, but now that I'm reading it on github, it seems rather unhelpful. I'm going to simply remove it and let the original TODO as it is